compiler-core/src/type_/tests/warnings.rs

```diff
@@ -781,19 +781,6 @@ pub const x = some_module.x
     );
 }
 
-#[test]
-fn no_unused_warnings_for_broken_code() {
-    let src = r#"
-pub fn main() {
-  let x = 1
-  1 + ""
-  x
-}"#;
-    let warnings = VectorWarningEmitterIO::default();
-    _ = compile_module("test_module", src, Some(Rc::new(warnings.clone())), vec![]).unwrap_err();
-    assert!(warnings.take().is_empty());
-}
-
```

Comment:
Since inference for binary operations is now fault tolerant this does also raise a warning complaining that `1 + ""` is unused. Can we accept this, or should it be special cased somehow to still not raise a warning?